File: backend/app/schemas/wallet.py
import uuid
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel

from app.models.enums import TransactionType


class LedgerEntryOut(BaseModel):
    id: uuid.UUID
    transaction_ref: str
    transaction_type: TransactionType
    amount: Decimal
    currency: str
    balance_before: Decimal
    balance_after: Decimal
    description: str | None
    created_at: datetime

    model_config = {"from_attributes": True}


class WalletSummary(BaseModel):
    available_balance: Decimal
    locked_balance: Decimal
    currency: str
    is_simulated: bool = True

File: frontend-next/src/types/wallet.ts
export type TransactionType =
  | 'TRADE_BUY'
  | 'TRADE_SELL'
  | 'POSITION_CLOSE'
  | 'ADMIN_CREDIT'
  | 'ADMIN_DEBIT'
  | 'DEPOSIT'
  | 'WITHDRAWAL'

export interface LedgerEntry {
  id: string
  transaction_ref: string
  transaction_type: TransactionType
  amount: string
  currency: string
  balance_before: string
  balance_after: string
  description: string | null
  created_at: string
}

export interface WalletSummary {
  available_balance: string
  locked_balance: string
  currency: string
  is_simulated: boolean
}
